internals/search.py

```diff
@@ -194,7 +195,7 @@ def _sort_by_total_order(result_id_list, total_order_ids):
 
 def process_query(
     user_query, sort_spec=None, show_unlisted=False, show_deleted=False,
-    start=0, num=100):
+    start=0, num=DEFAULT_RESULTS_PER_PAGE):
```

Comment:
now that user can decide `start` and `num`, we should check array out of index in  `paginated_id_list = sorted_id_list[start : start + num]`, at line 241. e.g.  sorted_id_list[min(start, len(paginated_id_list))  : min(start + num, len(paginated_id_list))

Reply:
Python does the right thing for us automatically.  If these values exceed the length of the list, the expression returns only up to the end of the list, or an empty list if appropriate.